src/cscbike.cpp

```diff
@@ -42,424 +45,421 @@ void cscbike::writeCharacteristic(uint8_t* data, uint8_t data_len, QString info,
-        if (avgP < 50)
+    if (Heart.value() > 0) {
+        int avgP = ((settings.value(QStringLiteral("power_hr_pwr1"), 200).toDouble() *
+                     settings.value(QStringLiteral("power_hr_hr2"), 170).toDouble()) -
+                    (settings.value(QStringLiteral("power_hr_pwr2"), 230).toDouble() *
+                     settings.value(QStringLiteral("power_hr_hr1"), 150).toDouble())) /
+                       (settings.value(QStringLiteral("power_hr_hr2"), 170).toDouble() -
+                        settings.value(QStringLiteral("power_hr_hr1"), 150).toDouble()) +
+                   (Heart.value() * ((settings.value(QStringLiteral("power_hr_pwr1"), 200).toDouble() -
+                                      settings.value(QStringLiteral("power_hr_pwr2"), 230).toDouble()) /
+                                     (settings.value(QStringLiteral("power_hr_hr1"), 150).toDouble() -
+                                      settings.value(QStringLiteral("power_hr_hr2"), 170).toDouble())));
+        if (avgP < 50) {
             avgP = 50;
+        }
         m_watt = avgP;
-        debug("Current Watt: " + QString::number(m_watt.value()));
+        emit debug(QStringLiteral("Current Watt: ") + QString::number(m_watt.value()));
     }
 
-    if(m_control->state() == QLowEnergyController::UnconnectedState)
-    {
+    if (m_control->state() == QLowEnergyController::UnconnectedState) {
         emit disconnected();
         return;
     }
 
-    if(initRequest)
-    {
+    if (initRequest) {
         initRequest = false;
-    }
-    else if(bluetoothDevice.isValid() &&
-       m_control->state() == QLowEnergyController::DiscoveredState //&&
-       //gattCommunicationChannelService &&
-       //gattWriteCharacteristic.isValid() &&
-       //gattNotify1Characteristic.isValid() &&
-       /*initDone*/)
-    {
+    } else if (bluetoothDevice.isValid() &&
+               m_control->state() == QLowEnergyController::DiscoveredState //&&
+                                                                           // gattCommunicationChannelService &&
+                                                                           // gattWriteCharacteristic.isValid() &&
+                                                                           // gattNotify1Characteristic.isValid() &&
+               /*initDone*/) {
         update_metrics(true, watts());
 
         // updating the treadmill console every second
-        if(sec1Update++ == (500 / refresh->interval()))
-        {
+        if (sec1Update++ == (500 / refresh->interval())) {
```

Comment:
in this case we shold use 500ms isn't it?

Reply:
It seems to me that sec1Update value is not used. Isn't it?

Reply:
@p3g4asus yeah it was just an example. I mean there is the same pseudocode in almost all the modules i guess

Reply:
Oh sorry. In this case I would use
```cpp
if(++sec1Update == (1000 / refresh->interval()))
```

Reply:
@p3g4asus it isn't what i meant. my question was about 500 vs 500ms using the std::chrono library :)

Reply:
@cagnulein the compiler won't let you do that, the resulting value of QTime::interval() is an int and not an std::chrono::milliseconds.

Reply:
There's this one though :
`std::chrono::milliseconds QTimer::intervalAsDuration() const`

Reply:
ok thanks for the clarification! I was trying only to split hairs in your awesome job @lifof  😆

Reply:
The pleasure is mine to contribute to such a great project :wink: 

This project solved a problem that was making me angry for a long time on that Elliptical of mine, that is the ability to use it with a custom HR belt instead of the one integrated in it's handles :laughing:  and Zwift is a great plus to that